src/ClientData/include/ClientData/TimerChain.h

```diff
@@ -57,6 +57,16 @@ class TimerBlock {
     return data_[idx];
   }
 
+  // Assuming timers are sorted, returns the first one which its end isn't smaller than min_ns.
```

Comment:
merely nitpicking: 

Assuming timers are sorted, returns the first one for which the end isn't smaller than min_ns.

or

Assuming timers are sorted, returns the first one for which the right interval border is not smaller than min_ns.

Reply:
or Assuming timers are sorted, returns the first one of which the end >= min_ns.

Reply:
I chose a modified version of the first proposed by daniel.